source/posix/atomic.c

```diff
@@ -0,0 +1,34 @@
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#if defined(__GNUC__) || defined(__clang__)
+
+    int aws_atomic_add(int *dst, int addend) {
+        return __sync_fetch_and_add(dst, addend);
```

Comment:
I considered this. Thing is the difference in user code is a plus or minus, added or dropped. It may be higher priority for us to have less code to maintain than to add in the other option.